src/3d/qgs3daxis.cpp

```diff
@@ -55,28 +58,23 @@ Qgs3DAxis::Qgs3DAxis( Qt3DExtras::Qt3DWindow *parentWindow,
   mTwoDLabelViewport = constructLabelViewport( parent3DScene, QRectF( 0.0f, 0.0f, 1.0f, 1.0f ) );
   mTwoDLabelViewport->setParent( mParentWindow->activeFrameGraph() );
 
-  Qgs3DAxisSettings s = mMapSettings->get3dAxisSettings();
-  setAxisViewportPosition( mAxisViewportSize, s.verticalPosition(), s.horizontalPosition() );
-  mMode = s.mode();
-
   connect( cameraCtrl, &QgsCameraController::cameraChanged, this, &Qgs3DAxis::onCameraUpdate );
   connect( mParentWindow, &Qt3DExtras::Qt3DWindow::widthChanged, this, &Qgs3DAxis::onAxisViewportSizeUpdate );
   connect( mParentWindow, &Qt3DExtras::Qt3DWindow::heightChanged, this, &Qgs3DAxis::onAxisViewportSizeUpdate );
 
-  onAxisViewportSizeUpdate();
-
   createAxisScene();
-
-  mMenu = new QMenu();
-  connect( mMenu, &QMenu::aboutToShow, this, &Qgs3DAxis::populateMenu );
+  onAxisViewportSizeUpdate();
 
   init3DObjectPicking();
 }
 
 Qgs3DAxis::~Qgs3DAxis()
 {
-  delete mMenu;
-  mMenu = nullptr;
+  if ( mMenu )
```

Comment:
by the way, it's not necessary to test `if ( mMenu )` before doing delete. Deleting a null pointer will not do anything...

Reply:
ok